src/persist-client/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(clippy::todo)]
+
+//! A proposal for alternate persist API, reinvented for platform.
+
+use std::marker::PhantomData;
+
+use differential_dataflow::difference::Semigroup;
+use mz_persist_types::{Codec, Codec64};
+use serde::{Deserialize, Serialize};
+use timely::progress::Timestamp;
+
+use crate::error::Error;
+use crate::read::ReadHandle;
+use crate::write::WriteHandle;
+
+pub mod error;
+pub mod read;
+pub mod write;
+
+// Notes
+// - Pretend that everything I've marked with Serialize and Deserialize instead
+//   has some sort of encode/decode API that uses byte slices, Buf+BufMut, or
+//   proto, depending on what we decide.
+// - PhantomData is a placeholder for internal state that users don't care
+//   about.
+
+// WIP
+// - Which of these need to be durably storable (for restarts) or transmittable
```

Comment:
seems like `Id`, `Location`, and `ReadHandle` need to be able to go over the network